index.js

```diff
@@ -28,6 +27,130 @@ var warn = function warn(message) {
 
 var replace = String.prototype.replace;
 
+// #### Pluralization methods
+// The string that separates the different phrase possibilities.
+var delimeter = '||||';
+
+// Mapping from pluralization group plural logic.
+var pluralTypes = {
+  arabic: function (n) {
+    // http://www.arabeyes.org/Plural_Forms
+    if (n < 3) { return n; }
+    if (n % 100 >= 3 && n % 100 <= 10) return 3;
+    return n % 100 >= 11 ? 4 : 5;
+  },
+  chinese: function () { return 0; },
+  german: function (n) { return n !== 1 ? 1 : 0; },
+  french: function (n) { return n > 1 ? 1 : 0; },
+  russian: function (n) {
+    if (n % 10 === 1 && n % 100 !== 11) { return 0; }
+    return n % 10 >= 2 && n % 10 <= 4 && (n % 100 < 10 || n % 100 >= 20) ? 1 : 2;
+  },
+  czech: function (n) {
+    if (n === 1) { return 0; }
+    return (n >= 2 && n <= 4) ? 1 : 2;
+  },
+  polish: function (n) {
+    if (n === 1) { return 0; }
+    return n % 10 >= 2 && n % 10 <= 4 && (n % 100 < 10 || n % 100 >= 20) ? 1 : 2;
+  },
+  icelandic: function (n) { return (n % 10 !== 1 || n % 100 === 11) ? 1 : 0; }
+};
+
+// Mapping from pluralization group to individual locales.
+var pluralTypeToLanguages = {
+  arabic: ['ar'],
+  chinese: ['fa', 'id', 'ja', 'ko', 'lo', 'ms', 'th', 'tr', 'zh'],
+  german: ['da', 'de', 'en', 'es', 'fi', 'el', 'he', 'hu', 'it', 'nl', 'no', 'pt', 'sv'],
+  french: ['fr', 'tl', 'pt-br'],
+  russian: ['hr', 'ru', 'lt'],
+  czech: ['cs', 'sk'],
+  polish: ['pl'],
+  icelandic: ['is']
+};
+
+function langToTypeMap(mapping) {
+  var ret = {};
+  forEach(mapping, function (langs, type) {
+    forEach(langs, function (lang) {
+      ret[lang] = type;
+    });
+  });
+  return ret;
+}
+
+// Trim a string.
+var trimRe = /^\s+|\s+$/g;
+function trim(str) {
```

Comment:
instead of adding a custom (not spec compliant) implementation here, can we use https://www.npmjs.com/package/string.prototype.trim ?